phonegap/phonegap-plugin-push
# Issue-2266: Add compatibility with katzer/cordova-plugin-local-notifications 0.9.0-beta.3

Add compatibility with katzer/cordova-plugin-local-notifications 0.9.0-beta.3

## Description
- Add code to save previous UNUserNotificationCenterDelegate and call it for all non-push notifications
- Similar changes have been made to katzer/cordova-plugin-local-notifications in [this pull request](https://github.com/katzer/cordova-plugin-local-notifications/pull/1514)

## Related Issue
#2266 

## Motivation and Context
This solves the problem of this plugin clobbering any previously assigned delegate to the `UNUserNotificationCenter`. As @wezzle mentioned, "this is something all plugins working with the notification center should do as there can only be one delegate set on the notification center."

## How Has This Been Tested?
1. Create a sample application with both cordova-plugin-local-notifications and phonegap-plugin-push installed
2. Schedule a local notification for several seconds in the future, background the app, and observe that it shows up in the notification center
3. Send a push notification while the app is in the foreground and observe that the `on('notification')` callback is executed
4. Send a push notification while the app is in the background and observe that it appears in the notification center

## Screenshots (if appropriate):

## Types of changes

## Checklist:
